sdk/monitor/Azure.Monitor.Ingestion/api/Azure.Monitor.Ingestion.netstandard2.0.cs

```diff
@@ -21,16 +21,16 @@ public enum ServiceVersion
     }
     public partial class UploadFailedEventArgs : Azure.SyncAsyncEventArgs
     {
-        public UploadFailedEventArgs(System.Collections.Generic.List<object> failedLogs, System.Exception exception, bool isRunningSynchronously, System.Threading.CancellationToken cancellationToken) : base (default(bool), default(System.Threading.CancellationToken)) { }
+        public UploadFailedEventArgs(System.Collections.Generic.IEnumerable<object> failedLogs, System.Exception exception, bool isRunningSynchronously, System.Threading.CancellationToken cancellationToken) : base (default(bool), default(System.Threading.CancellationToken)) { }
         public System.Exception Exception { get { throw null; } }
-        public System.Collections.Generic.IReadOnlyList<object> FailedLogs { get { throw null; } }
+        public System.Collections.Generic.IEnumerable<object> FailedLogs { get { throw null; } }
     }
-    public partial class UploadLogsOptions
+    public partial class UploadOptions
```

Comment:
This library will also include uploading metrics in the future which will also likely have upload options. So, we should use `UploadLogsOptions` for this and when we support metrics, we can have `UploadMetricsOptions`.

Reply:
If we want to use UploadLogsOptions, should we update the Upload method signature to be UploadLogs so the method and options class match? @tg-msft

Reply:
We decided to omit "Logs" from the method name because it would be redundant, given the name of the client class. We also have Monitor SDK UX study feedback to support this naming decision.

Reply:
Is the suggestion we'd have a separate client in the future?  Or add a new method called `UploadMetrics`?  The first option would potentially rename our options bag to something `LogsIngestionUploadOptions`.  The second would only require a longer name for the options bag of the future overload.

Reply:
Yes, we'd have a separate client for metrics (similar to what we have in Azure Monitor Query) called MetricsIngestionClient with `upload` methods. We could call this `UploadLogsOptions` or `LogsIngestionUploadOptions` but the former is shorter and doesn't have to include both "ingestion" and "upload" in the name.

Reply:
chatted offline - will rename options to LogsUploadOptions